prov/usnic/src/usdf_domain.c

```diff
@@ -362,3 +369,58 @@ int usdf_domain_getname(uint32_t version, struct usd_device_attrs *dap,
 	*name = buf;
 	return ret;
 }
+
+/* In pre-1.4 the domain name was NULL. This is unfortunate as it makes it
+ * difficult to tell whether providing a name was intended. In this case, it can
+ * be broken into 4 cases:
+ *
+ * 1. Version is greater than 1.4 and a non-NULL hint is provided. Just do a
+ *    string compare.
+ * 2. Version is greater than 1.4 and provided hint is NULL.  Treat this as
+ *    _valid_ as it could be an application requesting a 1.4 domain name but not
+ *    providing an explicit hint.
+ * 3. Version is less than 1.4 and a name hint is provided.  This should always
+ *    be _invalid_.
+ * 4. Version is less than 1.4 and name hint is NULL. This will always be
+ *    _valid_.
+ */
+bool usdf_domain_checkname(uint32_t version, struct usd_device_attrs *dap,
+			   char *hint)
+{
+	char *reference;
+	bool valid;
+	int ret;
+
+	USDF_DBG("checking domain name: version=%d, domain name='%s'\n",
+		 version, hint);
+
+	valid = false;
+	if (version) {
+		ret = usdf_domain_getname(version, dap, &reference);
+		if (ret < 0)
+			return false;
+
+		if (reference) {
```

Comment:
Could you put a comment here explaining that this is the `>=v1.4` case?  It took me a few moments to realize this (especially because the comment above the function describes everything in terms of `>=v1.4` and `<v1.4`).  Ditto for the `else` clause on this `if` block.